### PR TITLE
feat: Expandable List Row,  Accessibility Tweaks, Rotation Animation

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Add Expandable List Row examples + accessibility tweaks, as well as adds a Rotation/Animation component.

I can pull the Rotation/Animation component out of this PR, it just felt easier to add it here + then implement it throughout our repos as a follow-up PR.

### PR Checklist

- [x] Related to designs:
- [x] Related to JIRA ticket: gm-401
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.5-alpha.60ca6d.0<br/>@codecademy/gamut-labs@27.1.2-alpha.60ca6d.0<br/>@codecademy/gamut@39.1.2-alpha.60ca6d.0<br/>@codecademy/monaco@0.5.5-alpha.60ca6d.0<br/>@codecademy/styleguide@50.1.2-alpha.60ca6d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://617ff33b4c91283780028b32--gamut-preview.netlify.app>https://617ff33b4c91283780028b32--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/617ff33b4c91283780028b32>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.5-alpha.abf88b.0<br/>@codecademy/gamut-labs@27.1.2-alpha.abf88b.0<br/>@codecademy/gamut@39.1.2-alpha.abf88b.0<br/>@codecademy/monaco@0.5.5-alpha.abf88b.0<br/>@codecademy/styleguide@50.1.2-alpha.abf88b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6180165bf3d23b4a8623a9a3--gamut-preview.netlify.app>https://6180165bf3d23b4a8623a9a3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6180165bf3d23b4a8623a9a3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.6-alpha.9a7d95.0<br/>@codecademy/gamut-labs@28.0.1-alpha.9a7d95.0<br/>@codecademy/gamut@40.0.1-alpha.9a7d95.0<br/>@codecademy/monaco@0.5.6-alpha.9a7d95.0<br/>@codecademy/styleguide@50.1.3-alpha.9a7d95.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6182a25ba6bbdc13e03146ea--gamut-preview.netlify.app>https://6182a25ba6bbdc13e03146ea--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6182a25ba6bbdc13e03146ea>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.6-alpha.723d4d.0<br/>@codecademy/gamut-labs@28.0.1-alpha.723d4d.0<br/>@codecademy/gamut@40.0.1-alpha.723d4d.0<br/>@codecademy/monaco@0.5.6-alpha.723d4d.0<br/>@codecademy/styleguide@50.1.3-alpha.723d4d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6182aff226af0721cc84ce3d--gamut-preview.netlify.app>https://6182aff226af0721cc84ce3d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6182aff226af0721cc84ce3d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.7-alpha.333be4.0<br/>@codecademy/gamut-labs@28.0.2-alpha.333be4.0<br/>@codecademy/gamut@40.0.1-alpha.333be4.0<br/>@codecademy/monaco@0.5.6-alpha.333be4.0<br/>@codecademy/styleguide@50.1.4-alpha.333be4.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.8-alpha.eddbbd.0<br/>@codecademy/gamut-labs@28.0.3-alpha.eddbbd.0<br/>@codecademy/gamut@40.0.1-alpha.eddbbd.0<br/>@codecademy/monaco@0.5.6-alpha.eddbbd.0<br/>@codecademy/styleguide@50.1.5-alpha.eddbbd.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.8-alpha.a1ad65.0<br/>@codecademy/gamut-labs@28.0.3-alpha.a1ad65.0<br/>@codecademy/gamut@40.0.1-alpha.a1ad65.0<br/>@codecademy/monaco@0.5.6-alpha.a1ad65.0<br/>@codecademy/styleguide@50.1.5-alpha.a1ad65.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6182f4bd06808512b3bb0cdc--gamut-preview.netlify.app>https://6182f4bd06808512b3bb0cdc--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6182f4bd06808512b3bb0cdc>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.8-alpha.739187.0<br/>@codecademy/gamut-labs@28.0.3-alpha.739187.0<br/>@codecademy/gamut@40.0.1-alpha.739187.0<br/>@codecademy/monaco@0.5.6-alpha.739187.0<br/>@codecademy/styleguide@50.1.5-alpha.739187.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6182fdc258ab06210117dfbf--gamut-preview.netlify.app>https://6182fdc258ab06210117dfbf--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6182fdc258ab06210117dfbf>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.8-alpha.a510d1.0<br/>@codecademy/gamut-labs@28.0.3-alpha.a510d1.0<br/>@codecademy/gamut@40.0.1-alpha.a510d1.0<br/>@codecademy/monaco@0.5.6-alpha.a510d1.0<br/>@codecademy/styleguide@50.1.5-alpha.a510d1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6183013f6c6a9126ce2f5c25--gamut-preview.netlify.app>https://6183013f6c6a9126ce2f5c25--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6183013f6c6a9126ce2f5c25>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.8-alpha.c6a5ad.0<br/>@codecademy/gamut-labs@28.0.3-alpha.c6a5ad.0<br/>@codecademy/gamut@40.0.1-alpha.c6a5ad.0<br/>@codecademy/monaco@0.5.6-alpha.c6a5ad.0<br/>@codecademy/styleguide@50.1.5-alpha.c6a5ad.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.8-alpha.c1e9da.0<br/>@codecademy/gamut-labs@28.0.3-alpha.c1e9da.0<br/>@codecademy/gamut@40.0.1-alpha.c1e9da.0<br/>@codecademy/monaco@0.5.6-alpha.c1e9da.0<br/>@codecademy/styleguide@50.1.5-alpha.c1e9da.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6184137e19562d1e54d0a447--gamut-preview.netlify.app>https://6184137e19562d1e54d0a447--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6184137e19562d1e54d0a447>Deploy Logs</a></p>

File: packages/gamut/src/Animation/Rotation.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { motion } from 'framer-motion';
+import React from 'react';
+
+export interface RotationProps {
```
Comment (jakemhiller) [inline packages/gamut/src/Animation/Rotation.tsx:4]: do we want to extend this from the `motion.div` props, to allow animation changes? right now it's spreading the rest of the props but anything outside of these types will throw a TS error, right?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.9-alpha.240339.0<br/>@codecademy/gamut-labs@28.0.4-alpha.240339.0<br/>@codecademy/gamut@40.1.1-alpha.240339.0<br/>@codecademy/monaco@0.5.7-alpha.240339.0<br/>@codecademy/styleguide@50.1.6-alpha.240339.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61858ce92eabd2256a25907d--gamut-preview.netlify.app>https://61858ce92eabd2256a25907d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61858ce92eabd2256a25907d>Deploy Logs</a></p>

File: packages/gamut/src/Animation/Rotation.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+   * Height of container. Defaults to 16.†
+   */
+  height?: number;
+  /**
+   * Width of container. Defaults to 16.
+   */
+  width?: number;
+}
+
+export const Rotation: React.FC<RotationProps> = ({
```
Comment (sooinc) [inline packages/gamut/src/Animation/Rotation.tsx:25]: we can use this in so many places :D

File: packages/gamut/src/Animation/Rotation.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      transformOrigin: 'center',
+      alignSelf: 'center',
+      height,
+      width,
+    }}
+    variants={{
+      rotated: { rotate: degrees },
+      normal: { rotate: 0 },
+    }}
+    transition={{ duration: 0.2, ease: 'easeInOut' }}
```
Comment (sooinc) [inline packages/gamut/src/Animation/Rotation.tsx:45]: could we use the timing var from gamut-styles for duration?

File: packages/styleguide/stories/Atoms/Animation.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  }}
+/>
+
+<Canvas>
+  <Story name="Rotation">{(args) => <Rotation {...args} />}</Story>
+</Canvas>
+
+<PropsTable story="Rotation" />
+
+## Rotation Example
```
Comment (sooinc) [inline packages/styleguide/stories/Atoms/Animation.stories.mdx:29]: Very stoked for this addition!

File: packages/styleguide/stories/Organisms/Lists & Tables/List/examples.tsx [hunk 5]
```diff
@@ -173,12 +176,13 @@ export const NormalTemplate: React.FC<ListProps> = (args, { mode }) => (
 
 const sizes = ['content', 'sm', 'md', 'lg', 'xl'];
 
-export const ColumnTemplate: React.FC<ListProps> = (args) => {
+export const ColumnTemplate: React.FC<typeof List> = (args) => {
+  console.log(Box);
```
Comment (sooinc) [inline packages/styleguide/stories/Organisms/Lists & Tables/List/examples.tsx:180]: left a `console.log` here

File: packages/styleguide/stories/Organisms/Lists & Tables/List/index.stories.mdx [hunk 6]
```diff
@@ -277,6 +344,18 @@ Here's some examples of Lists to use as guides.
   </Story>
 </Canvas>
 
+<Canvas>
+  <Story name="Expanded - Button" args={{ variant: 'card', mode: 'dark' }}>
+    {ExpandedTemplateButtonClick.bind({})}
```
Comment (sooinc) [inline packages/styleguide/stories/Organisms/Lists & Tables/List/index.stories.mdx:349]: Why do we have to pass in an empty object here?

**Review by sooinc — APPROVED** [review]
Thanks so much for adding these features and having an ample amt of examples! I tested it out in monolith and it works well. Per our convo, portal-app needs to be updated with framer-motion dep. 

Looking forward to using this!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.10-alpha.dd65fb.0<br/>@codecademy/gamut-labs@28.0.5-alpha.dd65fb.0<br/>@codecademy/gamut@40.1.2-alpha.dd65fb.0<br/>@codecademy/monaco@0.5.8-alpha.dd65fb.0<br/>@codecademy/styleguide@50.1.7-alpha.dd65fb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6185b25043db9f0df09558b9--gamut-preview.netlify.app>https://6185b25043db9f0df09558b9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6185b25043db9f0df09558b9>Deploy Logs</a></p>

File: packages/gamut/src/Animation/Rotation.tsx (hunk 1 above)
Comment (dreamwasp) [reply packages/gamut/src/Animation/Rotation.tsx:4]: yeah definitely, will fix!

File: packages/gamut/src/Animation/Rotation.tsx (hunk 3 above)
Comment (dreamwasp) [reply packages/gamut/src/Animation/Rotation.tsx:45]: YES WE CAN, great call!!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.11-alpha.728354.0<br/>@codecademy/gamut-labs@28.0.6-alpha.728354.0<br/>@codecademy/gamut@40.1.3-alpha.728354.0<br/>@codecademy/monaco@0.5.9-alpha.728354.0<br/>@codecademy/styleguide@50.1.8-alpha.728354.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61896b83530b9a1c5b9b674e--gamut-preview.netlify.app>https://61896b83530b9a1c5b9b674e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61896b83530b9a1c5b9b674e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.12-alpha.8ddf2e.0<br/>@codecademy/gamut-labs@28.0.7-alpha.8ddf2e.0<br/>@codecademy/gamut@40.2.1-alpha.8ddf2e.0<br/>@codecademy/monaco@0.5.10-alpha.8ddf2e.0<br/>@codecademy/styleguide@50.1.9-alpha.8ddf2e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618aaaf38b38a61e30c483d9--gamut-preview.netlify.app>https://618aaaf38b38a61e30c483d9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618aaaf38b38a61e30c483d9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.13-alpha.5e604d.0<br/>@codecademy/gamut-labs@28.0.8-alpha.5e604d.0<br/>@codecademy/gamut@40.2.1-alpha.5e604d.0<br/>@codecademy/monaco@0.5.10-alpha.5e604d.0<br/>@codecademy/styleguide@50.1.10-alpha.5e604d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618ac5d51cfd403a768e25b9--gamut-preview.netlify.app>https://618ac5d51cfd403a768e25b9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618ac5d51cfd403a768e25b9>Deploy Logs</a></p>

File: packages/styleguide/stories/Organisms/Lists & Tables/List/index.stories.mdx (hunk 6 above)
Comment (sooinc) [reply packages/styleguide/stories/Organisms/Lists & Tables/List/index.stories.mdx:359]: ah got it, thank you for sharing!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.13-alpha.501850.0<br/>@codecademy/gamut-labs@28.0.8-alpha.501850.0<br/>@codecademy/gamut@40.2.1-alpha.501850.0<br/>@codecademy/monaco@0.5.10-alpha.501850.0<br/>@codecademy/styleguide@50.1.10-alpha.501850.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618adfea96d01d1a6bd0b4d3--gamut-preview.netlify.app>https://618adfea96d01d1a6bd0b4d3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618adfea96d01d1a6bd0b4d3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.13-alpha.00d6e5.0<br/>@codecademy/gamut-labs@28.0.8-alpha.00d6e5.0<br/>@codecademy/gamut@40.2.1-alpha.00d6e5.0<br/>@codecademy/monaco@0.5.10-alpha.00d6e5.0<br/>@codecademy/styleguide@50.1.10-alpha.00d6e5.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618af60e469b5c305e954804--gamut-preview.netlify.app>https://618af60e469b5c305e954804--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618af60e469b5c305e954804>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.13-alpha.d7c721.0<br/>@codecademy/gamut-labs@28.0.8-alpha.d7c721.0<br/>@codecademy/gamut@40.2.1-alpha.d7c721.0<br/>@codecademy/monaco@0.5.10-alpha.d7c721.0<br/>@codecademy/styleguide@50.1.10-alpha.d7c721.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618afeff308946009ae47e06--gamut-preview.netlify.app>https://618afeff308946009ae47e06--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618afeff308946009ae47e06>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.13-alpha.220de3.0<br/>@codecademy/gamut-labs@28.0.8-alpha.220de3.0<br/>@codecademy/gamut@40.2.1-alpha.220de3.0<br/>@codecademy/monaco@0.5.10-alpha.220de3.0<br/>@codecademy/styleguide@50.1.10-alpha.220de3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618b0162d627450e11060a2e--gamut-preview.netlify.app>https://618b0162d627450e11060a2e--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618b0162d627450e11060a2e>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.13-alpha.e3a611.0<br/>@codecademy/gamut-labs@28.0.8-alpha.e3a611.0<br/>@codecademy/gamut@40.2.1-alpha.e3a611.0<br/>@codecademy/monaco@0.5.10-alpha.e3a611.0<br/>@codecademy/styleguide@50.1.10-alpha.e3a611.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618b03acf246ca0cb800995b--gamut-preview.netlify.app>https://618b03acf246ca0cb800995b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618b03acf246ca0cb800995b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.13-alpha.1039d8.0<br/>@codecademy/gamut-labs@28.0.8-alpha.1039d8.0<br/>@codecademy/gamut@40.3.0-alpha.1039d8.0<br/>@codecademy/monaco@0.5.10-alpha.1039d8.0<br/>@codecademy/styleguide@50.2.0-alpha.1039d8.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://618bf74be4c13f56628a36e5--gamut-preview.netlify.app>https://618bf74be4c13f56628a36e5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/618bf74be4c13f56628a36e5>Deploy Logs</a></p>

**Pull request merged**